packages/client/hmi-client/src/router/index.ts

```diff
@@ -41,7 +46,17 @@ const router = createRouter({
 
 export function useCurrentRouter() {
 	return {
-		isCurrentRouteHome: computed(() => router.currentRoute.value.path === RoutePath.Home)
+		isCurrentRouteHome: computed(() => router.currentRoute.value.path === RoutePath.Home),
+		routerState: computed(() => {
+			const path = router.currentRoute.value.path;
+
+			// This maps into the navigation hierarchy :view/:viewId/:subView/:subViewId, eg
+			// - /projects/123/simulations/232
+			// - /projects/222/models/456
+			// - /docs/d123
+			const [, view, viewId, subView, subViewId] = path.split('/');
+			return { view, viewId, subView, subViewId };
```

Comment:
`vue-router` doesn't provide a better way to do this than `path.split('/')`?
Can we use RegEx to have at least better control on routing the user. I feel like returning an object seems so overkill.

Reply:
I think if you have canonical path-params you can sort of use vue-router to just tell you what the current ids are, but since there are some overlap it can be ambiguous.

I don't know what you mean by using regular expressions. Can you elaborate?

Reply:
like this: https://v3.router.vuejs.org/guide/essentials/dynamic-matching.html#dynamic-route-matching

Reply:
But I am using the matching patterns 😕 
```
	ModelView = '/projects/:projectId/model',
	SimulationView = '/projects/:projectId/simulation',
```

Reply:
yes, but this comment is outdated now